tests/UI/Component/MainControls/MainBarTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+require_once("libs/composer/vendor/autoload.php");
+require_once(__DIR__."/../../Base.php");
+
+use \ILIAS\UI\Component as C;
+use \ILIAS\UI\Implementation as I;
+use \ILIAS\UI\Implementation\Component\MainControls\Slate\Legacy;
+use \ILIAS\UI\Component\Signal;
+
+/**
+ * Tests for the Main Bar.
+ */
+class MainBarTest extends ILIAS_UI_TestBase
+{
+	public function setUp()
+	{
+		$sig_gen = 	new I\Component\SignalGenerator();
+		$this->factory = new I\Component\MainControls\Factory($sig_gen);
+		$this->button_factory = new I\Component\Button\Factory($sig_gen);
+		$this->icon_factory = new I\Component\Icon\Factory();
+
+		$this->mainbar = $this->factory->mainBar();
+	}
+
+	public function testConstruction()
+	{
+		$this->assertInstanceOf(
+			"ILIAS\\UI\\Component\\MainControls\\MainBar",
+			$this->mainbar
+		);
+	}
+
+	protected function getButton()
+	{
+		$symbol = $this->icon_factory->custom('', '');
+		return $this->button_factory->bulky($symbol,'TestEntry', '#');
+	}
+
+	protected function getSlate()
+	{
+		$mock = $this->getMockBuilder(Legacy::class)
+			->disableOriginalConstructor()
+			->setMethods(["transformToLegacyComponent"])
+			->getMock();
+
+		$mock->method('transformToLegacyComponent')->willReturn('content');
+		return $mock;
+	}
+
+	public function testAddEntry()
+	{
+		$btn = $this->getButton();
+		$mb = $this->mainbar->withAdditionalEntry('test', $btn);
+		$entries = $mb->getEntries();
+		$this->assertEquals($btn, $entries['test']);
+	}
+
+	public function testDisallowedEntry()
+	{
+		try {
+			$mb = $this->mainbar->withAdditionalEntry('test', 'wrong_param');
+		}
+		catch (\InvalidArgumentException $e) {
+			$this->assertTrue(true);
+		}
```

Comment:
Please use `this->expectException` instead. I once liked this syntax over the `@expectedException`, but the method convinced me to be better then the explicity catching.